lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -440,6 +461,39 @@ LogicalResult FIRRTLLowering::lowerBinOp(Operation *op) {
   return setLoweringTo<ResultOpType>(op, lhs, rhs);
 }
 
+/// lowerCmpOp extends each operand to the longest type, then performs the
+/// specified binary operator.
+template <typename ResultOpSignedType, typename ResultOpUnsignedType>
+LogicalResult FIRRTLLowering::lowerCmpOp(Operation *op, bool flip) {
+  // Extend the two operands to match the longest type.
+  Type resultType = builder->getIntegerType(1);
+  auto lhsFIRType =
+      op->getOperand(0).getType().cast<FIRRTLType>().getPassiveType();
+  auto lhsIntType = lhsFIRType.dyn_cast<IntType>();
+  auto rhsFIRType =
+      op->getOperand(1).getType().cast<FIRRTLType>().getPassiveType();
+  auto rhsIntType = rhsFIRType.dyn_cast<IntType>();
+  
+  if (!lhsIntType || !lhsIntType.hasWidth() || !rhsIntType || !rhsIntType.hasWidth())
+    return failure();
+  
+  Type cmpType = *lhsIntType.getWidth() < *rhsIntType.getWidth() ? rhsFIRType : lhsFIRType;
+
+  auto lhs = getLoweredAndExtendedValue(op->getOperand(0), cmpType);
+  auto rhs = getLoweredAndExtendedValue(op->getOperand(1), cmpType);
```

Comment:
The logic looks great, please wrap to 80 columns (with clang-format though) thanks!